nilearn/surface/surface.py

```diff
@@ -1014,3 +1014,235 @@ def check_surface(surface):
+        neighbors. A value of `-inf` results in no smoothing because the entire
+        weight is on the center, so each vertex is updated with its own value. A
+        value of `inf` results in each vertex being updated with the average of
+        its neighbors without including its own value.
+
+    match : { None | 'sum' | 'mean' | 'var' | 'dist' }, optional
+        What properties of the input data should be matched in the output data.
+        `None` indicates that the smoothed output should be
+        returned without transformation. If the value is `'sum'`, then the
+        output is rescaled to have the same sum as `surf_data`. If the value is
+        `'mean'`, then the output is shifted to match the mean of the input. If
+        the value is `'var'` or `'std'`, then the variance of the output is
+        matched. Finally, if the value iis `'dist'`, then the mean and the
+        variance are matched. Default is `'sum'` 
+    
+    Returns
+    -------
+    surf_data_smooth : array
+        The array of smoothed values at each vertex.
+
+    Examples
+    -------
+    >>> from nilearn import datasets, surface, plotting
+    >>> fsaverage = datasets.fetch_surf_fsaverage('fsaverage')
+    >>> white_left = surface.load_surf_mesh(fsaverage.white_left)
+    >>> curv = surface.load_surf_data(fsaverage.curv_left)
+    >>> curv_smooth = surface.smooth_surface_data(surface=white_left, surf_data=curv, iterations=50)
+    >>> plotting.plot_surf(white_left, surf_map = curv_smooth)
+
+    """
+    from scipy.sparse import diags
+    surface = load_surf_mesh(surface)
+    # First, calculate the center and surround weights for the
+    # center-surround knob.
+    t = center_surround_knob
+    center_weight = 1 / (1 + np.exp2(-t))
+    surround_weight = 1 - center_weight
+    if surround_weight == 0:
+        # There's nothing to do in this case.
+        return np.array(surf_data)
+    # Calculate the adjacency matrix either weighting by inverse distance or not weighting (ones)
+    if distance_weights:
+        matrix = compute_adjacency_matrix(surface, values='invlen')
+    else:
+        matrix = compute_adjacency_matrix(surface, values='ones')
+    
+    # If there are vertex weights, get them ready.
+    if vertex_weights:
+        w = np.array(vertex_weights)
+        w /= np.sum(w)
+    else:
+        w = np.ones(matrix.shape[0])
+    # We need to normalize the matrix columns, and we can do this now by
+    # normalizing everything but the diagonal to the surround weight, then
+    # adding the center weight along the diagonal.
+    colsums = matrix.sum(axis=1)
+    colsums = np.asarray(colsums).flatten()
+    matrix = matrix.multiply(surround_weight / colsums[:,None])
+
+    # Add in the diagonal.
+    matrix.setdiag(center_weight)
+    # Run the iteratioons of smooothing.
+    n = len(surf_data)
+    # data = np.hstack([np.reshape(surf_data, (n,-1)), w[:,None]])
```

Comment:
Is the following commented section not useful anymore? In this case could you remove it?